### PR TITLE
Doc extra

### PR DESCRIPTION
**Pull request by cbielow**
this PR adds some extra information for the user in multiple places:

1) amend our docs, for the zlib error of X!Tandem in #4817
2) write the search engine used in idXML files when calling FileInfo
3) some more stats when calling InternalCalibration (which calibration points were excluded and why)
4) allow MSGF+Adapter to pass a `conf` file with additional parameters

### CODE REVIEW

**Comment by cbielow** [comment]
tests pass.
Comments?

File: src/topp/MSGFPlusAdapter.cpp [hunk 1]
```diff
@@ -208,6 +219,8 @@ class MSGFPlusAdapter :
 
     registerFlag_("legacy_conversion", "Use the indirect conversion of MS-GF+ results to idXML via export to TSV. Try this only if the default conversion takes too long or uses too much memory.", true);
 
+    registerInputFile_("conf", "<file>", "", "Optional MSGF+ configuration file (passed as -conf <file> to MSGF+). See documentation for examples.", false, false);
```
Comment (timosachsenberg) [inline src/topp/MSGFPlusAdapter.cpp:222]: do we need to document here how deviating options set on command line and in config are handled?

Comment (cbielow) [reply src/topp/MSGFPlusAdapter.cpp:222]: I briefly looked at the msgf docs but no info.
We'd need to test that (hoping they do not change the behavior)

Comment (timosachsenberg) [reply src/topp/MSGFPlusAdapter.cpp:222]: one way would be to allow either conf or the other params

**Pull request merged**